src/rdkafka_telemetry_encode.c

```diff
@@ -0,0 +1,867 @@
+
+                if (is_per_broker_metric(rk, (int)metrics_to_encode[i])) {
+                        rd_kafka_broker_t *rkb;
+
+                        TAILQ_FOREACH(rkb, &rk->rk_brokers, rkb_link) {
+                                metrics[metric_idx] = rd_calloc(
+                                    1,
+                                    sizeof(
+                                        opentelemetry_proto_metrics_v1_Metric));
+                                data_points[metric_idx] = rd_calloc(
+                                    1,
+                                    sizeof(
+                                        opentelemetry_proto_metrics_v1_NumberDataPoint));
+                                serializeMetricData(
+                                    rk, rkb, &info[metrics_to_encode[i]],
+                                    &metrics[metric_idx],
+                                    &data_points[metric_idx],
+                                    &datapoint_attributes_key_values
+                                        [metric_idx],
+                                    metricValueCalculator,
+                                    &metric_names[metric_idx], true, now_ns);
+                                metric_idx++;
+                        }
+                        continue;
+                }
+
+                metrics[metric_idx] =
+                    rd_calloc(1, sizeof(opentelemetry_proto_metrics_v1_Metric));
+                data_points[metric_idx] = rd_calloc(
+                    1, sizeof(opentelemetry_proto_metrics_v1_NumberDataPoint));
+
+                serializeMetricData(
+                    rk, NULL, &info[metrics_to_encode[i]], &metrics[metric_idx],
+                    &data_points[metric_idx],
+                    &datapoint_attributes_key_values[metric_idx],
+                    metricValueCalculator, &metric_names[metric_idx], false,
+                    now_ns);
+                metric_idx++;
+        }
+
+        /* Send empty metrics blob if no metrics are matched */
+        if (total_metrics_count > 0) {
+                metrics_repeated.metrics = metrics;
+                metrics_repeated.count   = total_metrics_count;
+
+                scopeMetrics.metrics.funcs.encode = &encode_metric;
+                scopeMetrics.metrics.arg          = &metrics_repeated;
+
+
+                resource_metrics.scope_metrics.funcs.encode =
+                    &encode_scope_metrics;
+                resource_metrics.scope_metrics.arg = &scopeMetrics;
+
+                metrics_data.resource_metrics.funcs.encode =
+                    &encode_resource_metrics;
+                metrics_data.resource_metrics.arg = &resource_metrics;
+        }
+
+        status = pb_get_encoded_size(
+            &message_size, opentelemetry_proto_metrics_v1_MetricsData_fields,
+            &metrics_data);
+        if (!status) {
+                rd_kafka_dbg(rk, TELEMETRY, "RD_KAFKA_TELEMETRY_METRICS_INFO",
```

Comment:
Subcategory corresponding to the protocol message
```suggestion
                rd_kafka_dbg(rk, TELEMETRY, "PUSH",
```

Reply:
Updated